input/socket.js

```diff
@@ -0,0 +1,73 @@
+const dgram = require('dgram')
+const net = require('net')
+
+const { fromJS } = require('immutable')
+
+const defaultParse = s => fromJS(JSON.parse(s))
+
+function createTcpServer ({pipeline, name, port, handler}) {
+  return net
+    .createServer(socket => {
+      socket.on('data', data => {
+        data.toString().split('\n').forEach(line => {
+          if (line) {
+            handler(line)
+          }
+        })
+      })
+    })
+    .listen(port, err => {
+      if (err) {
+        pipeline.status(err, 'Cannot start: ' + err.message)
+      } else {
+        pipeline.status(null, `Listening on port ${port}.`)
+        console.log(`${name} listening on TCP port ${port}.`)
```

Comment:
Not directly linked to this PR but could the pipeline.status actually take care of logging such stuff? @benfle

Reply:
That would make sense, yes. Feel free to create an issue for it.